src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    static final double MAX_LENGTH = 25.0;
+    static final int AMOUNT_OF_FIGURES = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE.getColor(), 10);
+    }
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(AMOUNT_OF_FIGURES) + 1;
+        String color = colorSupplier.getRandomColor();
+        switch (randomFigure) {
+            case 1:
+                double radius = MAX_LENGTH * random.nextDouble();
+                return new Circle(color, radius);

```

Comment:
you may create additional method called `getRandomUnits()` and reuse it in all places:
```
private double getRandomUnits() {
   return MAX_LENGTH * random.nextDouble();
}
```
```suggestion
                return new Circle(color, getRandomUnits());

```